api/app/models/text.rb

```diff
@@ -28,7 +28,7 @@ class Text < ApplicationRecord
   acts_as_list scope: [:project_id, :category_id]
 
   # Associations
-  belongs_to :project, optional: true
+  belongs_to :project, optional: true, touch: true
```

Comment:
Updating a text will update the project.  This seemed like the only relationship that should do this, but let me know if you think there are others that should as well.